components/net/fetch/methods.rs

```diff
@@ -569,17 +569,17 @@ pub enum RangeRequestBounds {
 }
 
 impl RangeRequestBounds {
-    pub fn get_final(&self, len: Option<u64>) -> Result<RelativePos, ()> {
+    pub fn get_final(&self, len: Option<u64>) -> Option<RelativePos> {
```

Comment:
An error might help with diagnosing problems down the line. Could this just return an `Err` that's a `&'static str` explaining where in the process this failed? Even if we eventually throw those errors away they can help in debugging or be surfaced later.

Reply:
That makes a lot of sense, this way the debug information is there so it can be used if needed. I changed it back and added the error description on the last commit, thanks!